org-eclipse-che-jdt-ls-extension/jdt.ls.extension.core/src/main/java/org/eclipse/che/jdt/ls/extension/core/internal/pom/ReImportMavenProjectsHandler.java

```diff
@@ -136,36 +95,42 @@
   }
 
   /**
-   * Submits jobs to update given maven projects.
+   * Updates maven projects.
    *
    * @param projects list of projects to be updated
+   * @return list of jobs to update projects. It is needed for tests
    */
-  private static void submitUpdateJobs(List<IProject> projects) {
+  public static List<Job> updateProjects(List<IProject> projects) {
+    List<Job> updatedJobs = new ArrayList<>();
     for (IProject project : projects) {
-      projectsManager.updateProject(project, true);
+      Job job = JavaLanguageServerPlugin.getProjectsManager().updateProject(project, true);
+      job.addJobChangeListener(new JobChangedListener());
```

Comment:
We should add a listener here that captures all the context we need, which in our case is the project that is being updated.

Reply:
added project's uri to the listener.